tests/Silverback.Integration.Tests.E2E/Kafka/CallbacksTests.cs

```diff
@@ -151,6 +155,330 @@ public async Task EndpointsConfiguredCallback_MessagePublished()
+                .Run();
+
+            var kafkaTestingHelper = Host.ServiceProvider.GetRequiredService<IKafkaTestingHelper>();
+
+            var publisher = Host.ScopedServiceProvider.GetRequiredService<IEventPublisher>();
+            var callbackHandlerKafkaEndOfPartitionReached = (KafkaEndOfTopicPartitionReachedCallback)Host.ScopedServiceProvider
+                .GetServices<IBrokerCallback>()
+                .First(service => service is KafkaEndOfTopicPartitionReachedCallback);
+
+            await kafkaTestingHelper.WaitUntilAllMessagesAreConsumedAsync();
+
+            callbackHandlerKafkaEndOfPartitionReached.AllEndOfPartitionReachedCount.Should().Be(2);
+
+            await kafkaTestingHelper.Broker.Consumers[0].DisconnectAsync();
+
+            await publisher.PublishAsync(messageEventOne);
+            await publisher.PublishAsync(messageEventTwo);
+
+            await kafkaTestingHelper.Broker.Consumers[0].ConnectAsync();
+
+            await kafkaTestingHelper.WaitUntilAllMessagesAreConsumedAsync();
+
+            kafkaTestingHelper.Spy.InboundEnvelopes.Should().HaveCount(2);
+
+            callbackHandlerKafkaEndOfPartitionReached.AllEndOfPartitionReachedCount.Should().Be(4);
+
+            await kafkaTestingHelper.Broker.Consumers[0].DisconnectAsync();
+
+            await publisher.PublishAsync(messageEventOne);
+            await publisher.PublishAsync(messageEventTwo);
+
+            await kafkaTestingHelper.Broker.Consumers[0].ConnectAsync();
+
+            await kafkaTestingHelper.WaitUntilAllMessagesAreConsumedAsync();
+
+            kafkaTestingHelper.Spy.InboundEnvelopes.Should().HaveCount(4);
+            callbackHandlerKafkaEndOfPartitionReached.AllEndOfPartitionReachedCount.Should().Be(6);
+
+            await kafkaTestingHelper.Broker.Consumers[0].DisconnectAsync();
+
+            await publisher.PublishAsync(messageEventOne);
+            await publisher.PublishAsync(messageEventTwo);
+
+            await kafkaTestingHelper.Broker.Consumers[0].ConnectAsync();
+
+            await kafkaTestingHelper.WaitUntilAllMessagesAreConsumedAsync();
+
+            kafkaTestingHelper.Spy.InboundEnvelopes.Should().HaveCount(6);
+            callbackHandlerKafkaEndOfPartitionReached.AllEndOfPartitionReachedCount.Should().Be(8);
+
+            await kafkaTestingHelper.Broker.Consumers[0].DisconnectAsync();
+
+            await publisher.PublishAsync(messageEventOne);
+            await publisher.PublishAsync(messageEventTwo);
+
+            await kafkaTestingHelper.Broker.Consumers[0].ConnectAsync();
+
+            await kafkaTestingHelper.WaitUntilAllMessagesAreConsumedAsync();
+
+            kafkaTestingHelper.Spy.InboundEnvelopes.Should().HaveCount(8);
+            callbackHandlerKafkaEndOfPartitionReached.AllEndOfPartitionReachedCount.Should().Be(10);
+        }
+
+        [Fact]
+        public async Task KafkaEndOfTopicPartitionConfigured_PublishSeveralMessages_HandlerShouldBeInvokedForRightTopic()
```

Comment:
`..ForRightTopic`? You mean for the right partition? The test uses a single topic.

Anyway, would a test with  multiple topics be worth it in your opinion?

Reply:
You are right, of course. Changed it to ..ForRightPartition